src/hosted-buttons/utils.js

```diff
@@ -38,9 +40,43 @@ export const getMerchantID = (): string | void => {
   return merchantIds[0];
 };
 
+export const requestWithDPoP: RequestWithDPoP = async ({
+  url,
+  headers,
+  method,
+  body,
+}) => {
+  let accessToken, nonce;
+  if (!headers.Authorization?.match(/Basic/)) {
```

Comment:
this seems a little odd - thinking out loud here:
- I wanted to avoid rebuilding the headers in the three functions that call apis (`createAccessToken` `buildHostedButtonCreateOrder`, `buildHostedButtonCaptureOrder`) because:
- I would have to update the test cases for each of those (doubling the number of tests) but also I like how the existing tests are not updated to show that this new feature flag does not break existing functionality.
- this is feature flagged code that will be removed in the near(ish) future
- given that^, we can't pass `accessToken` to `requestWithDPoP` because `request` doesn't accept an `accessToken` parameter
- I thought parsing out the access token from the headers was less hacky than having if/else conditions in each of the api-calling methods

Reply:
Thinking out loud here, Can we encapsulate the `request` method inside another method `request_WithOut_DPoP` (we need a better name though) .
and `requestFn` can be either this or the other method. 
e.g. `const requestFn = enableDPoP ? requestWithDPoP : requestWithOutDPoP;` 

Under the hood both methods can invoke a memoized version of `createAccessToken` and then call `request`

Reply:
I like this suggestion! I tried it out but ran into the opposite issue, where `createAccessToken` doesn't have enough information to use `Authorization: Basic`. I have another idea (move this logic to `getHeaders` that I think will improve the readability and not require matching on the headers - stay tuned!

Reply:
I like this a lot better! a9299614 wdyt? (I abandoned the abstracted `requestWithDPoP` in favor of moving all the complexity to different test cases)